qiskit/pulse/instructions/__init__.py

```diff
@@ -10,7 +10,14 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""The ``instruction`` module holds the various ``Instruction`` s which are supported by
+r"""
+.. _pulse-insts:
+
+===============================================
+Instructions (:mod:`qiskit.pulse.instructions`)
+===============================================
+
+The ``instructions`` module holds the various :obj:`Instruction`\ s which are supported by
```

Comment:
Without a preceding `.`, Sphinx tries to match names within the `currentmodule` (`qiskit.pulse.instructions` here), then the parent module, and so on.  So it'll find `qiskit.pulse.instruction.Instruction` here, because `Instruction` is done by `automodule` later on here.  A lot of the problem we generally have is that our documentation generation is very implicit via autosummary cascading through a bunch of docstrings.  It's not clear what namespace something is defined in, much of the time...  Here it's much clearer what we mean, because there's an explicit `autoclass`.